extensions/pkg/controller/worker/genericactuator/actuator_delete.go

```diff
@@ -68,25 +69,19 @@ func (a *genericActuator) Delete(ctx context.Context, worker *extensionsv1alpha1
 		return errors.Wrapf(err, "marking all machines for forceful deletion failed")
 	}
 
-	// Get the list of all existing machine deployments.
-	existingMachineDeployments := &machinev1alpha1.MachineDeploymentList{}
-	if err := a.client.List(ctx, existingMachineDeployments, client.InNamespace(worker.Namespace)); err != nil {
-		return err
-	}
-
 	// Delete all machine deployments.
-	if err := a.cleanupMachineDeployments(ctx, logger, existingMachineDeployments, nil); err != nil {
-		return errors.Wrapf(err, "cleaning up machine deployments failed")
+	if err := a.client.DeleteAllOf(ctx, &machinev1alpha1.MachineDeployment{}, client.InNamespace(worker.Namespace)); err != nil {
+		return errors.Wrapf(err, "cleaning up all machine deployments failed")
 	}
 
 	// Delete all machine classes.
-	if err := a.cleanupMachineClasses(ctx, logger, worker.Namespace, workerDelegate.MachineClassList(), nil); err != nil {
-		return errors.Wrapf(err, "cleaning up machine classes failed")
+	if err := a.client.DeleteAllOf(ctx, workerDelegate.MachineClass(), client.InNamespace(worker.Namespace)); err != nil {
```

Comment:
One note:
`client.DeleteAllOf` should also work with the list type, see https://github.com/kubernetes-sigs/controller-runtime/blob/33f1d80ebeaa33c9e3d86c86c713ac5847dbe859/pkg/client/client_cache.go#L54-L57
Though, adding the `MachineClass` func and doing it this way is probably a bit cleaner...

Reply:
Hm, in my tests using the list type I got the following error:

```
{"level":"error","ts":"2020-11-17T17:45:40.281+0100","logger":"controller","msg":"Reconciler error","controller":"worker_controller","name":"worker","namespace":"shoot--foobar--aws2","error":"cleaning up all machine classes failed: object does not implement the Object interfaces","errorVerbose":"object does not implement the Object interfaces\n"}
```

Reply:
Hmm, ok. Then my assumption seems to be wrong...